[Bug]: test_subvolume_group_pinning failed with error missing key "'distributed'"

### Description
The test case tests/functional/storageclass/test_csi_subvolume_group_property.py::TestCSISubvolumeGroup::test_subvolume_group_pinning is failing with error
```
/home/jenkins/workspace/qe-deploy-ocs-cluster-prod/ocs-ci/tests/functional/storageclass/test_csi_subvolume_group_property.py:25: KeyError


During handling of the above exception, another exception occurred:


self = <tests.functional.storageclass.test_csi_subvolume_group_property.TestCSISubvolumeGroup object at 0x7f42432ef400>


@jira("DFBUGS-2759")
@polarion_id("OCS-5740")
def test_subvolume_group_pinning(self):
    """
    Test that verifies that the pinning value of CephFilesystemSubVolumeGroup is 1
    """

    result = run_oc_command("get CephFilesystemSubVolumeGroup -o yaml")
    yaml_dict = yaml.safe_load("\n".join(result))
    try:
        pinning_val = yaml_dict["items"][0]["spec"]["pinning"]["distributed"]
    except KeyError as e:
        err_msg = 'Pinning property not found, missing key "%s"' % str(e)
        logger.error(err_msg)

      raise Exception(err_msg)
E           Exception: Pinning property not found, missing key "'distributed'"



/home/jenkins/workspace/qe-deploy-ocs-cluster-prod/ocs-ci/tests/functional/storageclass/test_csi_subvolume_group_property.py:29: Exception
```

This is because the value is no longer present in the CephFilesystemSubVolumeGroup CR.
The bug https://issues.redhat.com/browse/DFBUGS-2759 is closed without change in the CR.
 
### Steps to Reproduce
1.Run the test case tests/functional/storageclass/test_csi_subvolume_group_property.py::TestCSISubvolumeGroup::test_subvolume_group_pinning

### Actual Behavior
Test failed because spec pinning.distributed is not present in CephFilesystemSubVolumeGroup CRs
### Expected Behavior
The test should check for the status value instead
### Impact (likelihood of reproduction, impact on the cluster and on other tests, etc.)
No other test is impacted. Cluster is not impacted.
### Screenshots (if applicable)

### Environment
- Test Suite(s): tier2
- Platform(s): Bare metal, any other
- Version(s):
single client - ODF 4.19
multiclient (provider mode) - ODF 4,18 and 4.19
- OS:

### Additional Context

--- a/tests/functional/storageclass/test_csi_subvolume_group_property.py
+++ b/tests/functional/storageclass/test_csi_subvolume_group_property.py
@@ -1,9 +1,10 @@
 import logging
-import yaml
 
+from ocs_ci.framework import config
 from ocs_ci.framework.testlib import ManageTest, tier2, skipif_external_mode
 from ocs_ci.framework.pytest_customization.marks import green_squad, jira, polarion_id
-from ocs_ci.helpers.performance_lib import run_oc_command
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.ocp import OCP
 
 logger = logging.getLogger(__name__)
 
@@ -19,21 +20,27 @@ class TestCSISubvolumeGroup(ManageTest):
         Test that verifies that the pinning value of CephFilesystemSubVolumeGroup is 1
         """
 
-        result = run_oc_command("get CephFilesystemSubVolumeGroup -o yaml")
-        yaml_dict = yaml.safe_load("\n".join(result))
-        try:
-            pinning = yaml_dict["items"][0]["status"]["pinning"]
-            pinning_key, pinning_val = pinning.strip().split("=")
-            assert pinning_key == "distributed", "Pinning must be 'distributed'"
-            pinning_val = int(pinning_val)
-        except KeyError as e:
-            err_msg = 'Pinning property not found, missing key "%s"' % str(e)
-            logger.error(err_msg)
-            raise Exception(err_msg)
+        subvolumegroup = OCP(
+            kind=constants.CEPHFILESYSTEMSUBVOLUMEGROUP,
+            namespace=config.ENV_DATA["cluster_namespace"],
+        ).get()
+        for subvolumegroup_info in subvolumegroup["items"]:
+            logger.info(
+                f"Checking {constants.CEPHFILESYSTEMSUBVOLUMEGROUP} {subvolumegroup_info['metadata']['name']}"
+            )
+            try:
+                pinning = subvolumegroup_info["status"]["info"]["pinning"]
+                pinning_key, pinning_val = pinning.strip().split("=")
+                assert pinning_key == "distributed", "Pinning must be 'distributed'"
+                pinning_val = int(pinning_val)
+            except KeyError as e:
+                err_msg = 'Pinning property not found, missing key "%s"' % str(e)
+                logger.error(err_msg)
+                raise Exception(err_msg)
 
-        if pinning_val != 1:
-            err_msg = f"Expected pinning value 1, got {pinning_val} instead"
-            logger.error(err_msg)
-            raise Exception(err_msg)
+            if pinning_val != 1:
+                err_msg = f"Expected pinning value 1, got {pinning_val} instead"
+                logger.error(err_msg)
+                raise Exception(err_msg)
 
-        logger.info("Pinning value found and it is 1, as expected")
+            logger.info("Pinning value found and it is 1, as expected")

